samza-core/src/main/scala/org/apache/samza/config/JobConfig.scala

```diff
@@ -101,6 +101,8 @@ object JobConfig {
   val DIAGNOSTICS_APPENDER_CLASS = "job.diagnostics.appender.class"
   val DEFAULT_DIAGNOSTICS_APPENDER_CLASS = "org.apache.samza.logging.log4j.SimpleDiagnosticsAppender"
 
+  val STREAM_PARTITION_MAPPER_FACTORY = "job.stream.partition.mapper.factory"
```

Comment:
Given the default functionality of the StreamPartitionMapper above, this config doesnt seem warranted.